x/evm/keeper/smart_contract_deployment.go

```diff
@@ -247,7 +247,7 @@ func (k Keeper) deploySmartContractToChain(ctx context.Context, chainInfo *types
 	}
 	uniqueID := generateSmartContractID(ctx)
 	k.createSmartContractDeployment(ctx, smartContract, chainInfo, uniqueID[:])
-	lastEventNonce, err := k.Gravity.GetLastObservedEventNonce(ctx, chainInfo.GetChainReferenceID())
+	lastEventNonce, err := k.Gravity.GetLastObservedGravityNonce(ctx, chainInfo.GetChainReferenceID())
 	if err != nil {
 		return fmt.Errorf("failed to get last observed event nonce: %w", err)
```

Comment:
Should we rename these to `lastGravityNonce` to keep consistency?
```suggestion
	lastGravityNonce, err := k.Gravity.GetLastObservedGravityNonce(ctx, chainInfo.GetChainReferenceID())
	if err != nil {
		return fmt.Errorf("failed to get last observed gravity nonce: %w", err)
```